packages/webapp/src/containers/Animals/Inventory/index.tsx

```diff
@@ -288,4 +340,43 @@ function AnimalInventory({
   );
 }
 
+export function ExpandableAnimalInventory(props: ExpandableAnimalInventoryProps) {
+  const { t } = useTranslation();
+  const { inventory } = useAnimalInventory();
```

Comment:
Following up on the idea I mentioned above of having different parent components for the different "versions" of the inventory, we could set `inventory` as a prop for the `AnimalInventory` container, and then call `useInventory` from the parent components. So we'd end up with something like:

```js
const AnimalsInventorySummary = () => {
  const { inventory, isLoading } = useAnimalInventory();
  const animalCount = ...;
  <ExpandableAnimalInventory
    ...
    expandedContent={
      <AnimalsInventory
        inventory={inventory}
        isLoading={isLoading}
        tableMaxHeight={...}
        tableSpacerRowHeight={...}
        ...
       />
    }
  />
}

const AnimalsInventoryTaskSelection = () => {
  const { inventory, isLoading } = useAnimalInventory();
  <AnimalsInventory
    inventory={inventory}
    isLoading={isLoading}
    tableMaxHeight={...}
    tableSpacerRowHeight={...}
    ...
   />
}
```
so the double call of `useAnimalInventory` would also be solved

Reply:
Let me know what you think of the newer refactor... its still not optimal. Theres just so much shared logic I found it hard to split away without making it look more complex.